### DISCUSSION TITLE
Startup on Login

### DISCUSSION THREAD

**Discussion by NiffirgkcaJ** [andyholmes/gnome-shell-extension-gsconnect#1895] [Q&A] [answered]
Hi! I'd like to ask if it's now supported by GSConnect to autostart at login time, like it was on GNOME 46? I manually added the `.desktop` file of GSConnect in the `/home/USER/.config/autostart/` folder.

While this has worked for some time. I think that Valent works better than GSConnect at autostarting. Is there a better way to force GSConnect to autostart more reliably?

**Comment by ferdnyc** [comment]
@NiffirgkcaJ Sorry for the delay in responding.

First question, are you using GSConnect with a GNOME Shell desktop login? If so, you _shouldn't_ need to set up any sort of autostarting for it — as long as the Shell extension is installed and enabled, it'll be started by the GNOME Shell process at launch.

If you're using GSConnect without the Shell extension (outside of GNOME Shell), that may not work as well. My (unmerged) changes from #1814 might help some in that regard.

I don't think the `.desktop` file would be of any use in autostarting GSConnect, though; my guess is that it would be looking to communicate with an already-running GSConnect, rather than starting it up. (The `.desktop` file is _mostly_ just there to identify GSConnect to the system, so that it has a localized name and icon when listed in the Notifications panel of the GNOME Settings, for example.)

**Comment by NiffirgkcaJ** [comment]
It is okay, and I see.

I use the extension from the GNOME Extensions store, and I thought the `.desktop` file would have any use since it's what Valent did.

**Reply by ferdnyc** [reply, marked as answer]
\*nod\* I've never personally worked with Valent, but I assume it's not built as a GNOME Shell extension; thus, it would need to be started by some process at login. For Shell extensions like GSConnect that's not necessary, as GNOME Shell itself will start all enabled extensions automatically.

**Reply by NiffirgkcaJ** [reply]
I just used it as a temporary replacement for GSConnect since the update of GNOME 47 broke it and is quite inconsistent with its behaviour, but now it seems to be working. Thank you so much for your work.

**Reply by ferdnyc** [reply]
Yeah, sorry about that. We got the v58 update (tagged for GNOME 47 compatibility, but also carrying fixes for a few issues identified in v57) out a bit later than we should've.
